cli/oyster-cvm/src/commands/build.rs

```diff
@@ -1,46 +1,97 @@
 use crate::types::Platform;
 use anyhow::{Context, Result};
-use std::process::Command;
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
 use tracing::info;
 
 pub fn build_oyster_image(
     platform: Platform,
     docker_compose: &str,
     docker_images: &[String],
     output: &str,
+    commit_ref: &str,
 ) -> Result<()> {
+    info!("Checking if the current user is in the nix trusted-users list");
+    check_trusted_user()?;
```

Comment:
why are we using this check here? what happens with nix command if this is not checked?

Reply:
Actually this was was added as the error wasn't readable if trusted user was failing, but with nix user would instantly know. 
But yeah, to make things more organized, we can add a dependency checker, which scans for all things necessary in the env for that specific command.

Reply:
removed from build, this will only run with doctor.